internal/interpreter/api.rs

```diff
@@ -148,6 +168,151 @@ impl Value {
             _ => ValueType::Other,
         }
     }
+
+    /// Create a `Value` from a JSON Value
+    #[cfg(feature = "json")]
+    pub fn from_json(t: &i_slint_compiler::langtype::Type, v: &serde_json::Value) -> Option<Self> {
+        use i_slint_compiler::langtype;
+
+        use smol_str::ToSmolStr;
+
+        match v {
+            serde_json::Value::Null => Some(Self::Void),
+            serde_json::Value::Bool(b) => Some((*b).into()),
+            serde_json::Value::Number(n) => Some(Self::Number(n.as_f64().unwrap_or(f64::NAN))),
+            serde_json::Value::String(s) => match t {
+                langtype::Type::Enumeration(e) => {
+                    let s = s.to_smolstr();
+                    if e.values.contains(&s) {
+                        Some(Self::EnumerationValue(e.name.to_string(), s.into()))
+                    } else {
+                        eprintln!("Warning: Unexpected value for enum '{}': {}", e.name, s);
+                        None
+                    }
+                }
+                langtype::Type::Color => {
+                    if let Some(c) = string_to_color(s) {
+                        Some(Self::Brush(i_slint_core::Brush::SolidColor(c)))
+                    } else {
+                        eprintln!("Warning: Failed to parse color: {}", s);
+                        None
+                    }
+                }
+                langtype::Type::String => Some(SharedString::from(s.as_str()).into()),
+                langtype::Type::Image => match Image::load_from_path(std::path::Path::new(s)) {
+                    Ok(image) => Some(image.into()),
+                    Err(_) => {
+                        eprintln!("Warning: Failed to load image from path: {}", s);
+                        None
+                    }
+                },
+                langtype::Type::Brush => {
+                    if s.starts_with('#') {
+                        if let Some(c) = string_to_color(s) {
+                            Some(Self::Brush(i_slint_core::Brush::SolidColor(c)))
+                        } else {
+                            eprintln!("Warning: Failed to parse color: {}", s);
```

Comment:
Not great to use eprintln from a library. We should maybe use debug_log instead. Or maybe just ignore it?

Reply:
I turned the entire thing into `TryFrom` and have porper error reporting now.